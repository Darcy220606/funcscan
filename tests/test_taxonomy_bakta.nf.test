nextflow_pipeline {

    name "Test pipeline: NFCORE_FUNCSCAN"
    script "main.nf"
    tag "pipeline"
    tag "nfcore_funcscan"
    tag "test_taxonomy_bakta"

    test("test_taxonomy_bakta") {

        when {
            params {
                outdir = "$outputDir"
                run_taxa_classification = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml").exists() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },

                { assert snapshot(
                    file("$outputDir/amp/ampir/sample_1/sample_1.ampir.tsv").text.contains("NODE_882919_length_258_cov_0.935961_1"),
                    file("$outputDir/amp/ampir/sample_1/sample_1.ampir.faa").text.contains("NODE_882919_length_258_cov_0.935961_1"),
                    file("$outputDir/amp/ampir/sample_2/sample_2.ampir.tsv").text.contains("NODE_882919_length_258_cov_0.935961_1"),
                    file("$outputDir/amp/ampir/sample_2/sample_2.ampir.faa").text.contains("NODE_882919_length_258_cov_0.935961_1"),
                    file("$outputDir/amp/amplify/sample_1/sample_1.amplify.tsv").text.contains("NODE_882919_length_258_cov_0.935961_1"),
                    file("$outputDir/amp/amplify/sample_2/sample_2.amplify.tsv").text.contains("NODE_882919_length_258_cov_0.935961_1"),
                    path("$outputDir/amp/macrel/sample_1.macrel/sample_1.macrel.smorfs.faa.gz"),
                    path("$outputDir/amp/macrel/sample_2.macrel/sample_2.macrel.smorfs.faa.gz"),
                    path("$outputDir/amp/macrel/sample_1.macrel/sample_1.macrel.all_orfs.faa.gz"),
                    path("$outputDir/amp/macrel/sample_2.macrel/sample_2.macrel.all_orfs.faa.gz"),
                    path("$outputDir/amp/macrel/sample_1.macrel/sample_1.macrel.prediction.gz"),
                    path("$outputDir/amp/macrel/sample_2.macrel/sample_2.macrel.prediction.gz"),
                    path("$outputDir/amp/macrel/sample_1.macrel/README.md"),
                    path("$outputDir/amp/macrel/sample_2.macrel/README.md"),
                    path("$outputDir/amp/macrel/sample_1.macrel/sample_1.macrel_log.txt"),
                    path("$outputDir/amp/macrel/sample_2.macrel/sample_2.macrel_log.txt"),
                    path("$outputDir/arg/rgi/sample_1/sample_1.txt"),
                    path("$outputDir/arg/rgi/sample_2/sample_2.txt"),
                    file("$outputDir/arg/abricate/sample_1/sample_1.txt").text.contains("RESISTANCE"),
                    file("$outputDir/arg/abricate/sample_2/sample_2.txt").text.contains("COVERAGE_MAP"),
                    path("$outputDir/arg/fargene/sample_1/class_b_1_2/results_summary.txt"),
                    file("$outputDir/arg/fargene/sample_2/class_b_3/results_summary.txt").text.contains("class_B_3.hmm"),
                    file("$outputDir/arg/fargene/sample_2/tet_efflux/results_summary.txt").text.contains("tet_efflux.hmm"),
                    file("$outputDir/bgc/antismash/sample_2/sample_2.gbk").text.contains("##antiSMASH-Data-START##"),
                    file("$outputDir/bgc/antismash/sample_2/sample_2.log").text.contains("antiSMASH status: SUCCESS"),
                    path("$outputDir/bgc/gecco/sample_2/sample_2.genes.tsv"),
                    path("$outputDir/bgc/gecco/sample_2/sample_2.features.tsv"),
                    file("$outputDir/reports/ampcombi2/sample_2/sample_2_ampcombi.tsv").text.contains("NODE_515831_length_303_cov_1.532258_1"),
                    ).match() },
                { assert new File("$outputDir/amp/hmmer_hmmsearch/sample_1/sample_1_mybacteriocin.hmmer_hmmsearch.txt.gz").exists() },
                { assert new File("$outputDir/amp/hmmer_hmmsearch/sample_2/sample_2_mybacteriocin.hmmer_hmmsearch.txt.gz").exists() },
                { assert new File("$outputDir/reports/ampcombi2/ampcombi_complete_summary_taxonomy.tsv.gz").exists() },
                { assert new File("$outputDir/reports/hamronization_summarize/hamronization_combined_report.tsv.gz").exists() },
                { assert new File("$outputDir/bgc/hmmer_hmmsearch/sample_2/sample_2_ToyB.txt.gz").exists() }
            )
        }
    }
}
