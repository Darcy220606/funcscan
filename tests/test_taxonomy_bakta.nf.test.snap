{
    "test_taxonomy_bakta": {
        "content": [
            true,
            true,
            true,
            true,
            true,
            true,
            "sample_1.macrel.smorfs.faa.gz:md5,1b5e2434860e635e95324d1804a3be7b",
            "sample_2.macrel.smorfs.faa.gz:md5,38108b5cdfdc2196afe67418b9b04682",
            "sample_1.macrel.all_orfs.faa.gz:md5,86f6b3b590d1b22d9c5aa164f8a14080",
            "sample_2.macrel.all_orfs.faa.gz:md5,fdb384925af50ecade05dccaff68afd8",
            "sample_1.macrel.prediction.gz:md5,0c4b16e0838be56e012b99169863a168",
            "sample_2.macrel.prediction.gz:md5,440deffd6b6d9986ce098e44c66db9ae",
            "README.md:md5,fa3706dfc95d0538a52c4d0d824be5fb",
            "README.md:md5,fa3706dfc95d0538a52c4d0d824be5fb",
            "sample_1.macrel_log.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
            "sample_2.macrel_log.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
            "sample_1.txt:md5,fce130af51f93cccfc09ddaf9caf623f",
            "sample_2.txt:md5,fce130af51f93cccfc09ddaf9caf623f",
            true,
            true,
            "results_summary.txt:md5,f69af9b0e18a0cfc934eb18b7e4bffe1",
            true,
            true,
            true,
            true,
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-05T09:31:30.025771"
    }
}